### PR TITLE
Allow mutual authentication to be required for TLS server side

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -8207,6 +8207,48 @@ int wolfSSL_update_keys(WOLFSSL* ssl)
...
+ * Server side only.
+ *
+ * ctx  The SSL/TLS CTX object.
+ * req  1 to indicate required and 0 when not.
+ * returns BAD_FUNC_ARG when ctx is NULL, SIDE_ERROR when not a server and
+ * 0 on success.
+ */
+int wolfSSL_CTX_mutual_auth(WOLFSSL_CTX* ctx, int req)
+{
+    if (ctx == NULL || !IsAtLeastTLSv1_3(ctx->method->version))
```
Comment (dgarske) [inline src/tls13.c]: @SparkiDev : How difficult would this be to extend to older TLS versions as well? Seems like a desirable server side feature / option.

Comment (SparkiDev) [reply src/tls13.c]: Good idea!
After checking the TLS 1.2 spec, it is an option to send an empty certificate and ignore or complain about it.
I'll change the code.

**Comment by toddouska** [comment]
Please explain.  Our example server has required Mutual Auth by default for several years with WOLFSSL_VERIFY_PEER.  The -d option disables Mutual Auth.

**Comment by SparkiDev** [comment]
TLS 1.3 code does not error by default when mutual authentication is asked but not certificate is sent.
TLS 1.2 closes with an error and can be configured to not care.
This code change allows the user to choose to have an alert sent when the server asks for the client certificate and no certificate is returned as per the specs. TLS 1.2 specifies a handshake failure and TLS 1.3 specifies a certificate required alert.

**Comment by toddouska** [comment]
TLS 1.2 already allowed the server to not request a client cert (the default), the server to request a client cert and fail if not sent, and the the server to request a client cert and continue without failure when not sent.  Why did this need to be changed?

**Comment by kaleb-himes** [comment]
There was a support item where a customer noticed that if they loaded a cert for the client but no private key and did not specify -d to disable mutual auth the connection would still succeed. I sent Sean a note to see why the server was ignoring the fact that the wrong (or in this case no) private key was loaded. It appeared mutual auth was not happening as expected. These changes stemmed from that report.

- K

**Comment by SparkiDev** [comment]
TLS 1.3 didn't fail. So a changes was needed.
TLS 1.2 doesn't send an alert, it does now. I can undo this if you want.

**Comment by toddouska** [comment]
Thanks for clarifying the TLS 1.2 mutual auth change as alert only.  Please resolve server.c conflict and I'll merge.

**Pull request merged**